plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaRecordSet.java

```diff
@@ -123,7 +127,7 @@ public RecordCursor cursor()
         private KafkaRecordCursor()
         {
             topicPartition = new TopicPartition(split.getTopicName(), split.getPartitionId());
-            kafkaConsumer = consumerFactory.create();
+            kafkaConsumer = consumerFactory.create(connectorSession);
```

Comment:
nit : Can we wrap it as a `Supplier` of `KafkaConsumer` so that we don't have to pass `ConnectorSession` here ?

Reply:
I'm not sure I like this idea. Having an object created here directly through the factory method has one advantage: it's clear who should close it. In this case `KafkaRecordCursor` will close it when the cursor closes. Whereas passing a supplier is a bit different use case and have different meaning. Who should close the obtained through supplier? The cursor? I'm not convinced that it's the one which controls the lifetime of the object.